### ISSUE TITLE
Training with WavCaps dataset causes loss to diverge.

### ISSUE THREAD

**Issue by ribes14** [CPJKU/dcase2025_task6_baseline#1]
Hi there,

I've been experimenting with training the baseline system using different dataset combinations, and I've encountered a significant issue when including the WavCaps dataset alongside Clotho and AudioCaps.

When training only with Clotho and AudioCaps, the training process behaves as expected — the loss steadily decreases and the evaluation metrics are in line with the results presented in the task. However, when WavCaps is added, the training loss exhibits a concerning pattern: at a certain point, the loss spikes back up to the initial value seen at the very start of training, and then it remains stuck there for the rest of the training process.

This behavior is visually evident in the following training loss plot where the green curve indicates training with Clotho, AudioCaps, and WavCaps and the pink one training with just Clotho and AudioCaps.

![Image](https://github.com/user-attachments/assets/d4b69fc5-f687-4f3e-9bed-0d01e27a8e10)

Furthermore, the evaluation metrics after training with WavCaps included are significantly worse across the board. The evaluation chart shows how degraded the performance becomes — all key metrics drop drastically compared to training without WavCaps.

![Image](https://github.com/user-attachments/assets/765d7b28-4732-4d26-b9e9-49fb90744fcf)

Would appreciate any insights or guidance on how to approach this.

Thanks!

**Comment by OptimusPrimus** [comment]
Hey!

Thanks for pointing this out! 
It looks like there is a numerical issue at some point during the training; I'll investigate this to make the training more stable. 

As a workaround for now: run the experiment with the suggested seed (--seed=967251; not sure if it is completely reproducible though), increase the numerical precision, or modify the learning rate.

Best,
Paul

**Comment by OptimusPrimus** [comment]
Can you provide some more details (like runtime environment, seed, etc. ) so that I can reproduce this issue?

**Comment by ribes14** [comment]
Hi Paul,

Thanks for your quick response!

I went ahead and tried running the training with the seed you suggested (--seed=967251), but unfortunately, I still observed the same behavior as before — the training loss curve followed the same pattern as the previous green curve (where the loss increases and gets stuck mid-training).

For context, here are the arguments with  I executed the training script:

```
--logging
--data_path /data/
--batch_size 16
--batch_size_eval 16
--train
--test
--wavcaps
--audiocaps
--ablate_clean_setup

```
I suspect the issue might be related to how I downloaded the WavCaps dataset. Due to storage constraints, I didn’t use the training script’s built-in downloading mechanism. Instead, I used the `download_wavcaps_mp3()` function manually to fetch the data. I noticed the `download_wavcaps() ` function is present at training script but commented out, so I assumed it wasn’t necessary to use.

That said, I’ll double-check whether the data I downloaded matches the expected format and contents. I’ll also try experimenting with the learning rate and other parameters as you suggested to see if I can get more stable behavior.

In case it helps with reproducibility, I’m running everything inside a Docker container. Here’s the Dockerfile I’m using:

```
FROM nvidia/cuda:12.1.1-cudnn8-runtime-ubuntu22.04

RUN apt-get update && apt-get install -y --no-install-recommends \
    python3.11 \
    python3-pip \
    python3.11-distutils \
    p7zip-full \
    && ln -s /usr/bin/python3.11 /usr/local/bin/python \
    && rm -rf /var/lib/apt/lists/*

RUN update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.11 1

RUN pip3 install --upgrade pip && \
    pip3 install torch torchvision torchaudio 

COPY . /app

RUN pip3 install -r requirements.txt
RUN pip3 install ray[tune]
```
Let me know if there's anything else I can share that would help with debugging. I’ll keep experimenting on my own as well and report back if I find anything useful!

Best,

Rubén
